lib/document.js

```diff
@@ -631,10 +646,18 @@ Document.prototype._saveHook = function(docToSave, saveAll, savedModel, callback
   var p = new Promise(function(resolve, reject) {
     // Steps:
     // - Save belongsTo
+    // - Save pivot
     // - Save this
     // - Save hasOne, hasMany and hasAndBelongsToMany docs
     // - Save links
 
+    // Save pivot table
+    var firstPromise;
```

Comment:
I did try to save it with links; somehow I couldn't (I don't remember why; but saving the pivot here was working. Maybe we should investigate this).